index.js

```diff
@@ -1,114 +1,193 @@
+   * @param {String} indexUrl The path to the HTML or url, or a markdown file
+   *   with 'md' or 'markdown' extension.
+   * @param output The name of the file to export to.  If the extension is
+   *   '.png' then a PNG image will be generated instead of a PDF.
+   */
+  function render (indexUrl, output) {
+    var requestedURL = url.parse(indexUrl)
+    var win = launchBrowserWindow()
+    setSessionCookies(argv.cookies, requestedURL, win)
+    loadURL(win, indexUrl)
+    win.webContents.on('did-finish-load', function () {
+      setTimeout(
+        generateOutput.bind(this, win, output),
+        argv.outputWait)
+    })
+  }
 
-  function isMarkdown (input) {
-    var ext = path.extname(input)
-    return ext.indexOf('md') > 0 || ext.indexOf('markdown') > 0
+  /**
+   *
+   * @param {String} cookies - ';' delimited cookies, '=' delimited name/value
+   *   pairs
+   * @param {URL} requestedURL - URL Object
+   * @param {BrowserWindow} win - The electron browser window
+   */
+  function setSessionCookies (cookies, requestedURL, win) {
+    if (cookies) {
+      cookies.split(';').forEach(function (c) {
+        var nameValue = c.split('=')
+        var cookie = {
+          url: requestedURL.protocol + '//' + requestedURL.host,
+          name: nameValue[0],
+          value: nameValue[1]
+        }
+        win.webContents.session.cookies.set(cookie, function (err) {
+          if (err) {
+            console.log(err)
+          }
+        })
+      })
+    }
   }
 
-  if (isMarkdown(input)) {
-    var opts = {}
+  /**
+   *
+   * @returns {electron.BrowserWindow}
+   */
+  function launchBrowserWindow () {
+    function pdfToPixels (inches) {
+      return Math.floor(inches * HTML_DPI)
+    }
 
-    if (customCss) {
-      opts.customCss = customCss
+    var pageDimensions = {
+      'A3': {x: pdfToPixels(11.7), y: pdfToPixels(16.5)},
+      'A4': {x: pdfToPixels(8.3), y: pdfToPixels(11.7)},
+      'A5': {x: pdfToPixels(5.8), y: pdfToPixels(8.3)},
+      'Letter': {x: pdfToPixels(8.5), y: pdfToPixels(11)},
+      'Legal': {x: pdfToPixels(8.5), y: pdfToPixels(14)},
+      'Tabloid': {x: pdfToPixels(11), y: pdfToPixels(17)}
     }
 
-    // if given a markdown, render it into HTML and return the path of the HTML
-    input = markdownToHTMLPath(input, opts, function (err, tmpHTMLPath) {
-      if (err) {
-        console.error('Parse markdown file error', err)
-        app.quit()
-      }
+    var pageDim = pageDimensions[argv.pageSize]
+    pageDim = argv.landscape ? {x: pageDim.y, y: pageDim.x} : pageDim
 
-      var indexUrl = wargs.urlWithArgs(tmpHTMLPath, {})
-      render(indexUrl, output)
-    })
-  } else {
-    var indexUrl = wargs.urlWithArgs(input, {})
-    render(indexUrl, output)
-  }
-}
+    var defaultOpts = {
+      width: pageDim.x,
+      height: pageDim.y,
+      enableLargerThanScreen: true,
+      show: false
+    }
+
+    var browserConfig = _.extend(defaultOpts, JSON.parse(argv.browserConfig || '{}'))
```

Comment:
Might worth to add a `try catch` statement here to exist the process and throw an error message if user give an invalid JSON.